modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/endpoint/SchedulerRestService.java

```diff
@@ -1088,53 +1126,20 @@ public Response getEventsAsXml(@QueryParam("agent") String device, @QueryParam("
   public Response getConflictingEventsXml(@QueryParam("agent") String device, @QueryParam("rrule") String rrule,
           @QueryParam("start") Long startDate, @QueryParam("end") Long endDate, @QueryParam("duration") Long duration,
           @QueryParam("timezone") String timezone) throws UnauthorizedException {
-    if (StringUtils.isBlank(device) || startDate == null || endDate == null) {
-      logger.info("Either agent, start date or end date were not specified");
-      return Response.status(Status.BAD_REQUEST).build();
-    }
-
-    RRule rule = null;
-    if (StringUtils.isNotBlank(rrule)) {
-      if (duration == null || StringUtils.isBlank(timezone)) {
-        logger.info("Either duration or timezone were not specified");
-        return Response.status(Status.BAD_REQUEST).build();
-      }
-
-      try {
-        rule = new RRule(rrule);
-        rule.validate();
-      } catch (Exception e) {
-        logger.info("Unable to parse rrule {}: {}", rrule, getMessage(e));
-        return Response.status(Status.BAD_REQUEST).build();
-      }
-
-      if (!Arrays.asList(TimeZone.getAvailableIDs()).contains(timezone)) {
-        logger.info("Unable to parse timezone: {}", timezone);
-        return Response.status(Status.BAD_REQUEST).build();
-      }
-    }
-
-    Date start = new DateTime(startDate).toDateTime(DateTimeZone.UTC).toDate();
-
-    Date end = new DateTime(endDate).toDateTime(DateTimeZone.UTC).toDate();
-
     try {
-      List<MediaPackage> events;
-      if (StringUtils.isNotBlank(rrule)) {
-        events = service.findConflictingEvents(device, rule, start, end, duration, TimeZone.getTimeZone(timezone));
-      } else {
-        events = service.findConflictingEvents(device, start, end);
-      }
+      List<MediaPackage> events = getConflictingEvents(device, rrule, startDate, endDate, duration, timezone);
       if (!events.isEmpty()) {
         return Response.ok(MediaPackageParser.getArrayAsXml(events)).build();
       } else {
         return Response.noContent().build();
       }
+    } catch (IllegalArgumentException iae) {
+      return Response.status(Status.BAD_REQUEST).build();
     } catch (UnauthorizedException e) {
-      throw e;
+      return Response.status(Status.UNAUTHORIZED).build();
```

Comment:
I agree with your hesitation on changing this API. I notice that some of the other endpoints do a "throw e". So, maybe it would be better not to change it.